sbi/inference/posteriors/score_posterior.py

```diff
@@ -0,0 +1,393 @@
+            x_shape: Shape of a single simulator output. If passed, it is used to check
+                the shape of the observed data and give a descriptive error.
+            enable_transform: Whether to transform parameters to unconstrained space
+                during MAP optimization. When False, an identity transform will be
+                returned for `theta_transform`.
+        """
+
+        check_prior(prior)
+
+        super().__init__(
+            device=device,
+            x_shape=x_shape,
+        )
+
+        potential_fn_gradient, theta_transform = (
+            score_estimator_based_potential_gradient(
+                score_estimator=score_estimator,
+                prior=prior,
+                x_o=None,
+                enable_transform=enable_transform,
+            )
+        )
+
+        device = device if device is not None else potential_fn_gradient.device
+
+        self.prior = prior
+        self.score_estimator = score_estimator
+        self.potential_fn_gradient = potential_fn_gradient
+        self.theta_transform = theta_transform
+
+        self.max_sampling_batch_size = max_sampling_batch_size
+
+        self._purpose = """It samples from the diffusion model given the \
+            score_estimator."""
+
+    def sample(
+        self,
+        sample_shape: Shape = torch.Size(),
+        x: Optional[Tensor] = None,
+        predictor: Union[str, Predictor] = "euler_maruyama",
+        corrector: Optional[Union[str, Corrector]] = None,
+        predictor_params: Optional[Dict] = None,
+        corrector_params: Optional[Dict] = None,
+        steps: int = 500,
+        ts: Optional[Tensor] = None,
+        max_sampling_batch_size: int = 10_000,
+        sample_with: Optional[str] = None,
+        show_progress_bars: bool = True,
+    ) -> Tensor:
+        r"""Return samples from posterior distribution $p(\theta|x)$.
+
+        Args:
+            sample_shape: Shape of the samples to be drawn.
+            x: Deprecated - use `.set_default_x()` prior to `.sample()`.
+            method: Method to use for sampling. Currently, only "euler_maruyma" is
+                supported.
+            steps: Number of steps to take for the Euler-Maruyama method.
+            max_sampling_batch_size: Maximum batch size for sampling.
+            sample_with: Deprecated - use `.build_posterior(sample_with=...)` prior to
+                `.sample()`.
+            show_progress_bars: Whether to show a progress bar during sampling.
+        """
+
+        num_samples = torch.Size(sample_shape).numel()
+
+        x = self._x_else_default_x(x)
+        x = reshape_to_batch_event(x, self.score_estimator.condition_shape)
+        self.potential_fn_gradient.set_x(x)
+
+        max_sampling_batch_size = (
+            self.max_sampling_batch_size
+            if max_sampling_batch_size is None
+            else max_sampling_batch_size
+        )
+
+        if sample_with is not None:
```

Comment:
I think we can already drop the `sample_with` for this function, instead of raising a warning from v0.18.0

Reply:
agree, but let's not have this in this PR